From 18ec6bd6f839c6f47e3bcf04e7e8e39e3115a010 Mon Sep 17 00:00:00 2001
From: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
Date: Thu, 14 Dec 2017 13:29:46 +0530
Subject: [PATCH 0995/1074] Soundwire: Fix out of bounds access to port

In cnl_sdw_alloc_port() the sdw->port can be accessed at
CNL_SDW_MAX_PORTS index and will lead to out of bound access.

Fix out of bound access by correcting the for loop condition.

Change-Id: Ieff95701c79233ce030b6ab5e0a5915e6e59e9b5
Review-by: Vaibhav Agarwal <vaibhav.agarwal@intel.com>
Signed-off-by: Pankaj Bharadiya <pankaj.laxminarayan.bharadiya@intel.com>
---
 drivers/sdw/sdw_cnl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/sdw/sdw_cnl.c b/drivers/sdw/sdw_cnl.c
index 2e9773571b9b..857530f9e6c4 100644
--- a/drivers/sdw/sdw_cnl.c
+++ b/drivers/sdw/sdw_cnl.c
@@ -784,7 +784,7 @@ struct cnl_sdw_port *cnl_sdw_alloc_port(struct sdw_master *mstr, int ch_count,
 	sdw = sdw_master_get_drvdata(mstr);
 
 	mutex_lock(&sdw->stream_lock);
-	for (i = 1; i <= CNL_SDW_MAX_PORTS; i++) {
+	for (i = 1; i < CNL_SDW_MAX_PORTS; i++) {
 		if (sdw->port[i].allocated == false) {
 			port = &sdw->port[i];
 			port->allocated = true;
-- 
2.14.4

